codex.nim

```diff
@@ -67,6 +67,9 @@ when isMainModule:
     # permissions are insecure.
     quit QuitFailure
 
+  if config.prover() and not(checkAndCreateDataDir((config.circuitDir).string)):
```

Comment:
Minor nitpick: it looks a bit weird that we create dir for circuit when we expect it to already be there (clouds intent a bit IMHO).